tests/unit/s2n_shutdown_test.c

```diff
@@ -328,5 +328,95 @@ int main(int argc, char **argv)
         EXPECT_SUCCESS(s2n_io_pair_close_one_end(&io_pair, S2N_CLIENT));
     };
 
+    /* Test: s2n_shutdown_write */
```

Comment:
Wondering if we need any tests on how s2n_shutdown_write interacts with s2n_shutdown. Can you add a call to s2n_shutdown() to one of these tests just to make sure you can call both?

Reply:
That's covered by one of the self-talk s2n_alerts_protocol_test tests ("Sending a closure alert"). But I can add a simpler unit test version to s2n_shutdown_test.